app/update.ts

```diff
@@ -1,9 +1,20 @@
-import {canvas, frameLength} from 'app/gameConstants'
+import {canvas, context, frameLength} from 'app/gameConstants'
 import {state} from 'app/state';
 
 function update() {
     for (const object of state.world.objects) {
         object.update(state);
+        
+        // If the nexus is destroyed, stop update function
+        // Pan world.camera to nexus, change background color (gray) and return.
+        if (object.objectType === "nexus" && object.health <= 0){
+            console.log("nexus dead!!!");
```

Comment:
You can remove this logging now, and it looks like the fillStyle on the next line is also unused.